eva_submission/assembly_taxonomy_insertion.py

```diff
@@ -0,0 +1,183 @@
+from lxml import etree
+import requests
+
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query
+from ebi_eva_common_pyutils.logger import logging_config as log_cfg
+
+
+logger = log_cfg.get_logger(__name__)
+
+
+def download_xml_from_ena(accession):
+    """Download and parse XML from ENA using any accession or id (e.g. assembly or taxonomy)"""
+    try:  # catches any kind of request error, including non-20X status code
+        response = requests.get(f'https://www.ebi.ac.uk/ena/browser/api/xml/{accession}')
```

Comment:
Does this confer any advantages over data/view endpoint? I remember that some of the older accessions were not available (at the time of writing the original script) via the browser/api endpoint.

Reply:
Thanks, I wasn't aware of that!  To be honest I think this was just the api I could find [documentation](https://ena-docs.readthedocs.io/en/latest/retrieval/programmatic-access.html) for, it seemed a bit simpler and I spot-checked a couple requests to make sure they were the same.

If you happen to remember some of the older accessions it didn't work for I can double-check, if not I'll just change it back to data/view just to be safe.

Reply:
Easiest way to test this would be to compare results for the list of all assembly accession (EVA + dbSNP-imported) we already have in our accessioning database (because not finding one of those assemblies was what led to our subsequent conversation with ENA about the APIs). @andresfsilva was collecting this list for EVA stats so I am guessing he may be able to provide it.

Reply:
I would leave it for now. These functions should probably live in [eva-common-pyutils](https://github.com/EBIvariation/eva-common-pyutils) along with their Genbank equivalent. I would suggest to look into the API's completeness at that point.

Reply:
Fair enough, but in this case I think I'd rather go back to the one we know works and is already being used in the existing script... it's a quick change.